### PR TITLE
Refactor startup check before running any watch

### PR DESCRIPTION
**Pull request by jimmidyson**

### CODE REVIEW

File: src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java [hunk 1]
```diff
@@ -180,4 +169,30 @@ public static synchronized void buildAdded(Build build) throws IOException {
...
+          Long.parseLong(b2.getMetadata().getAnnotations().get(OPENSHIFT_ANNOTATIONS_BUILD_NUMBER))
+        );
+      }
+    });
+
+    for (Build b : builds) {
+      try {
+        buildAdded(b);
+      } catch (IOException e) {
+        e.printStackTrace();
```
Comment (scoheb) [inline src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java]: change to  log statement?

File: src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java [hunk 2]
```diff
@@ -114,6 +116,28 @@ private void configChange() {
         OpenShiftUtils.initializeOpenShiftClient(server);
         this.namespace = getNamespaceOrUseDefault(namespace, getOpenShiftClient());
 
+        logger.info("Waiting for Jenkins to be started");
```
Comment (scoheb) [inline src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java]: I am wondering why you need to do this here? Can you not wait until all jobs are loaded? Take a look at this: http://javadoc.jenkins.io/hudson/model/listeners/ItemListener.html#onLoaded()

Comment (scoheb) [inline src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java]: Why do you have to wait until Jenkins is ready for this task? Does it not suffice to wait until all jobs are loaded? See http://javadoc.jenkins.io/hudson/model/listeners/ItemListener.html#onLoaded()

Comment (jimmidyson) [inline src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java]: Good question. TBH I'm not 100% sure that everything is in the right state when `ItemListener.onLoaded` is called. The current way may not be the "best" or most correct way to do it but we know it works. I've raised fabric8io/openshift-jenkins-sync-plugin#109 to track it & test it out once I get some time.

File: src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java (hunk 1 above)
Comment (jimmidyson) [inline src/main/java/io/fabric8/jenkins/openshiftsync/BuildWatcher.java]: Done.

File: src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java (hunk 2 above)
Comment (scoheb) [inline src/main/java/io/fabric8/jenkins/openshiftsync/GlobalPluginConfiguration.java]: Sounds good to me!

**Pull request merged**